airbyte-integrations/connectors/source-lever/source_lever/source.py

```diff
@@ -0,0 +1,267 @@
+
+class Tags(LeverStream):
+    primary_key = "text"
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "tags"
+
+class Sources(LeverStream):
+    primary_key = "text"
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "sources"
+    
+class RequisitionFields(LeverStream):
+    primary_key = "text"
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "requisition_fields"
+
+
+class ArchiveReasons(LeverStream):
+    primary_key = "id"
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "archive_reasons"
+
+
+# TODO: Basic incremental stream
+class IncrementalLeverStream(LeverStream, ABC):
+    """
+    TODO fill in details of this class to implement functionality related to incremental syncs for your connector.
+         if you do not need to implement incremental sync for any streams, remove this class.
+    """
+
+    # TODO: Fill in to checkpoint stream reads after N records. This prevents re-reading of data if the stream fails for any reason.
+    state_checkpoint_interval = None
+
+    @property
+    def cursor_field(self) -> str:
+        """
+        TODO
+        Override to return the cursor field used by this stream e.g: an API entity might always use created_at as the cursor field. This is
+        usually id or date based. This field's presence tells the framework this in an incremental stream. Required for incremental.
+
+        :return str: The name of the cursor field.
+        """
+        return []
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Override to determine the latest state after reading the latest record. This typically compared the cursor_field from the latest record and
+        the current state and picks the 'most' recent cursor. This is how a stream's state is determined. Required for incremental.
+        """
+        return {}
+
+
+# Basic Sub streams using Opportunity id
+class OpportunitySubStream(LeverStream, ABC):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        
+    # def __init__(self, start_date: str, **kwargs):
+    #     super().__init__(**kwargs)
+    #     self._start_date = start_date
+    
+    def path(self, stream_slice: Mapping[str, any] = None, **kwargs) -> str:
+        return f"opportunities/{stream_slice['opportunity_id']}/{self.name}"
+
+    def stream_slices(self, **kwargs) -> Iterable[Optional[Mapping[str, Any]]]:
```

Comment:
Is it permitted to type `Any` as a more specific type?  Or must it match `Any`?

Reply:
Resolved this. It's permitted to type `Any`. Not being loose here, this is generated from the Airbyte script